find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
+  max = nil
```

Comment:
the only cases in which max array returns nil are where max_index is zero or negative. There is explicit messaging for max_index==0, but nothing for negative.  Maybe return nil in both those cases (with explicit messaging) and initialize max later in the else condition?
```suggestion
```